WordPress/Classes/ViewRelated/NUX/Login2FAViewController.swift

```diff
@@ -219,6 +242,13 @@ extension Login2FAViewController {
             // Invalid verification code.
             displayError(message: NSLocalizedString("Whoops, that's not a valid two-factor verification code. Double-check your code and try again!",
                                                     comment: "Error message shown when an incorrect two factor code is provided."))
+        } else if err.domain == "WordPressComOAuthError" && err.code == WordPressComOAuthError.invalidTwoStepCode.rawValue {
+            // Invalid 2FA during social login
+            if let newNonce = (error as NSError).userInfo[WordPressComOAuthClient.WordPressComOAuthErrorNewNonceKey] as? String {
+                loginFields.nonceInfo?.updateNonce(with: newNonce)
+            }
+            displayError(message: NSLocalizedString("Whoops, that's not a valid two-factor verification code. Double-check your code and try again!",
```

Comment:
Maybe we could define the error string in a `let` before the if block to avoid some duplication?